### PR TITLE
Fix file-descriptor leak

### PR DESCRIPTION
**Pull request by IljaN**
Clone client transport from http.DefaultTransport instead of creating a zeroed transport struct.
With an empty timeout-struct all timeout-values are zeroed disabling some timeouts and TTLs.

See https://golang.org/src/net/http/transport.go#L194 and following.

Closes https://github.com/owncloud/product/issues/100

### CODE REVIEW

**Comment by individual-it** [comment]
@jasson99 please test this fix

**Comment by labkode** [comment]
@IljaN nice catch, for long term I think we should avoid to rely on the default transport and always create it explicitly without relying on package variables.

**Comment by labkode** [comment]
Merge after @jasson99 validates.

**Comment by IljaN** [comment]
nofail

Hatch 0.25, 150 Concurrent

File: pkg/rhttp/client.go [hunk 1]
```diff
@@ -24,9 +24,10 @@ import (
...
 	"net/http"
 
+	"go.opencensus.io/plugin/ochttp"
+
 	"github.com/cs3org/reva/pkg/token"
 	"github.com/pkg/errors"
-	"go.opencensus.io/plugin/ochttp"
 )
 
 // GetHTTPClient returns an http client with open census tracing support.
```
Comment (refs) [inline pkg/rhttp/client.go:33]: is the TODO still necessary?

**Comment by IljaN** [comment]
> ...for long term I think we should avoid to rely on the default transport and always create it explicitly without relying on package variables.

Yes agree! We just need to find or tune the right values for our use-case. For now I assumed that the go-lang selected defaults should suite us. Any proposals?

**Comment by jasson99** [comment]
@IljaN I tested with this fix and it works now.

**Pull request merged**